### PR TITLE
lightning: support gbk

### LINKED ISSUE #31293: lightning fails to import table CHARSET=gbk
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Create tidb cluster with below parameters:
```
new_collations_enabled_on_first_bootstrap = true
[experimental]
enable-new-charset = true
```
2. Create table with charset gbk
```
create database gbktest; 
use gbktest;
create table gbk1(a int primary key, b varchar(20) charset gbk) collate gbk_bin; 
insert into gbk1 value (1, "GBK测试");
```
3. Use Dumpling to export the data
```
./dumpling  -h "172.16.6.217" -P "4000" -u "root" --output "local:///home/tools_test_data/gbktest" --filetype "csv" --rows "200" --threads "20"
```
4. Use lighting to import the data

### 2. What did you expect to see? (Required)
Import should succeed.

### 3. What did you see instead (Required)
Import failed due to: Error: [parser:1115]Unknown character set: 'gbk'
```
[root@centos76_vm v5.4.0]# ./tidb-lightning --tidb-host=172.16.6.217 --tidb-port=4000 --backend=local --sorted-kv-dir=/home/sorted-kvs -d 'local:///home/tools_test_data/gbktest'
Verbose debug logs will be written to /tmp/lightning.log.2022-01-04T17.51.57+0800

Error: [parser:1115]Unknown character set: 'gb18030'
tidb lightning encountered error:  [parser:1115]Unknown character set: 'gb18030'

```
### 4. What is your TiDB version? (Required)
[root@centos76_vm v5.4.0]# ./tidb-lightning -V
Release Version: v5.4.0
Git Commit Hash: 974b5784adbbd47d14659916d47dd986effa7b4e
Git Branch: heads/refs/tags/v5.4.0
Go Version: go1.16.4
UTC Build Time: 2022-01-04 07:44:08
Race Enabled: false

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #31293

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/log/log.go [hunk 1]
```diff
@@ -220,12 +221,13 @@ func (task *Task) End(level zapcore.Level, err error, extraFields ...zap.Field)
...
 		verb = " failed"
 		extraFields = nil
+		errDetail = zap.Error(err)
 	}
 	if ce := task.WithOptions(zap.AddCallerSkip(1)).Check(level, task.name+verb); ce != nil {
 		ce.Write(append(
 			extraFields,
 			zap.Duration("takeTime", elapsed),
-			ShortError(err),
+			errDetail,
```
Comment (kennytm) [inline br/pkg/lightning/log/log.go:230]: i don't think this change should be contained in this pr

**Review by kennytm — APPROVED** [review]
rest lgtm

File: br/pkg/lightning/log/log.go (hunk 1 above)
Comment (glorv) [reply br/pkg/lightning/log/log.go]: rollbacked😂

**Review by gozssky — APPROVED** [review]

**Comment by glorv** [comment]
closed in favor of https://github.com/pingcap/tidb/pull/31291

**Pull request closed without merging**